clusterloader2/pkg/measurement/util/wait_for_pvcs.go

```diff
@@ -0,0 +1,92 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+// WaitForPVCOptions is an options used by WaitForPVCs methods.
+type WaitForPVCOptions struct {
+	Selector            *ObjectSelector
+	DesiredPVCCount     int
+	EnableLogging       bool
+	CallerName          string
+	WaitForPVCsInterval time.Duration
+}
+
+// WaitForPVCs waits till disire nuber of PVCs is running.
+// PVCs are be specified by namespace, field and/or label selectors.
+// If stopCh is closed before all PVCs are running, the error will be returned.
+func WaitForPVCs(clientSet clientset.Interface, stopCh <-chan struct{}, options *WaitForPVCOptions) error {
+	ps, err := NewPVCStore(clientSet, options.Selector)
+	if err != nil {
+		return fmt.Errorf("PVC store creation error: %v", err)
+	}
+	defer ps.Stop()
+
+	pvcs := ps.List()
+	var oldPVCs []*corev1.PersistentVolumeClaim
+	pvcsStatus := ComputePVCStartupStatus(pvcs, options.DesiredPVCCount)
+	scaling := uninitialized
+
+	switch {
+	case len(pvcs) == options.DesiredPVCCount:
+		scaling = none
+	case len(pvcs) < options.DesiredPVCCount:
+		scaling = up
+	case len(pvcs) > options.DesiredPVCCount:
+		scaling = down
+	}
+
+	for {
+		select {
+		case <-stopCh:
+			return fmt.Errorf("timeout while waiting for %d PVCs to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found bound",
+				options.DesiredPVCCount, options.Selector.Namespace, options.Selector.LabelSelector, options.Selector.FieldSelector, pvcsStatus.Bound)
+		case <-time.After(options.WaitForPVCsInterval):
+			diff := DiffPVCs(oldPVCs, pvcs)
+			deletedPVCs := diff.DeletedPVCs()
+			if scaling != down && len(deletedPVCs) > 0 {
+				klog.Errorf("%s: %s: %d PVCs disappeared: %v", options.CallerName, options.Selector.String(), len(deletedPVCs), strings.Join(deletedPVCs, ", "))
+				klog.Infof("%s: %v", options.CallerName, diff.String(sets.NewString()))
+			}
+			addedPVCs := diff.AddedPVCs()
+			if scaling != up && len(addedPVCs) > 0 {
+				klog.Errorf("%s: %s: %d PVCs appeared: %v", options.CallerName, options.Selector.String(), len(deletedPVCs), strings.Join(deletedPVCs, ", "))
+				klog.Infof("%s: %v", options.CallerName, diff.String(sets.NewString()))
+			}
+			if options.EnableLogging {
+				klog.Infof("%s: %s: %s", options.CallerName, options.Selector.String(), pvcsStatus.String())
+			}
+			// We wait until there is a desired number of PVCs bound and all other PVCs are pending.
+			if len(pvcs) == (pvcsStatus.Bound+pvcsStatus.Pending) && pvcsStatus.Bound == options.DesiredPVCCount {
+				return nil
+			}
+			oldPVCs = pvcs
+			pvcs = ps.List()
```

Comment:
I don't like this, because you are processing pvs that were there WaitForPVCsInterval ago (which sounds wrong).
But I don't want to block this PR for a long - I will send out a PR to fix it and unify with wait_for_pods a bit.